addons/escoria-core/game/core-scripts/esc/esc_action_manager.gd

```diff
@@ -49,6 +69,119 @@ var action_state = ACTION_INPUT_STATE.AWAITING_VERB_OR_ITEM \
 		setget set_action_input_state
 
 
+# Run a generic action
+#
+# #### Parameters
+#
+# - action: type of the action to run
+# - params: Parameters for the action
+# - can_interrupt: if true, this command will interrupt any ongoing event 
+# before it is finished
+func do(action: int, params: Array = [], can_interrupt: bool = false) -> void:
+	if escoria.current_state == escoria.GAME_STATE.DEFAULT:
+		match action:
+			ACTION.BACKGROUND_CLICK:
```

Comment:
I'm thinking about whether we shouldn't use a "do" command but rather implement methods like background_click or such and call those from the other scripts.

Pro: Params would be typed, jumping direct to code
Con: Loose possibilities to quickly add ne "do" actions.

What do you think? @BHSDuncan @StraToN

Reply:
That would complexify a bit the game script implementation though, as that change would add 5 new methods (1 per input type) instead of one "do()". I am not against nor in favor, but I'm eager to discuss this.

Reply:
All good points above. In my opinion, I think the main question is whether we'll ever need to add additional actions to the `do` method. If we don't, then it's probably ok as it is and would save complicating things. If we do, however, then the `do` method will quickly balloon and there's a decent chance we'd need additional "helper" parameters like `can_interrupt`; in that case, we're probably better off introducing a bit of complexity to save making the code less maintainable and less readable.

Just my thoughts. :)

Reply:
Well, we could just leave it like it is and if its complexity blows up, we refactor then. Always good to only do the stuff we need right now. So maybe I'll withdraw my comment. :)

Reply:
Agreed. Time to put a clamp on the scope creep for this ticket. Good to take note of, though.